src/sentry/uptime/consumers/results_consumer.py

```diff
@@ -74,6 +79,38 @@ class UptimeResultProcessor(ResultProcessor[CheckResult, UptimeSubscription]):
     def get_subscription_id(self, result: CheckResult) -> str:
         return result["subscription_id"]
 
+    def check_and_update_regions(self, subscription: UptimeSubscription):
+        # Run region checks and updates roughly once an hour
+        chance_to_run = subscription.interval_seconds / timedelta(hours=1).total_seconds()
+        if random.random() < chance_to_run:
```

Comment:
nit: flip around to guard clause

```suggestion
        if random.random() >= chance_to_run:
            return
```